### PR TITLE
Wip rgw d4n

### PR DESCRIPTION
**Pull request by samarahu**
Updates for several review comments on main PR

### CODE REVIEW

File: src/rgw/rgw_sal.cc [hunk 1]
```diff
@@ -118,7 +118,7 @@ rgw::sal::Driver* DriverManager::init_storage_provider(const DoutPrefixProvider*
   rgw::sal::Driver* driver{nullptr};
 
   if (cfg.store_name.compare("rados") == 0) {
-    driver = newRadosStore(&io_context);
+    driver = newRadosStore(io_context);
```
Comment (pritha-srivastava) [inline src/rgw/rgw_sal.cc]: You will have to make the same change in rgw_sal_rados.cc

File: src/rgw/rgw_sal.cc [hunk 2]
```diff
@@ -63,7 +63,7 @@ extern rgw::sal::Driver* newPOSIXDriver(rgw::sal::Driver* next);
 #endif
 extern rgw::sal::Driver* newBaseFilter(rgw::sal::Driver* next);
 #ifdef WITH_RADOSGW_D4N
-extern rgw::sal::Driver* newD4NFilter(rgw::sal::Driver* next, boost::asio::io_context* io_context);
+extern rgw::sal::Driver* newD4NFilter(rgw::sal::Driver* next, boost::asio::io_context& io_context);
```
Comment (pritha-srivastava) [inline src/rgw/rgw_sal.cc]: same change in rgw_sal_d4n.cc

Comment (samarahu) [reply src/rgw/rgw_sal.cc]: Do you mean updating the `newD4NFilter` method to match the function prototype here?

File: src/rgw/rgw_sal.cc (hunk 1 above)
Comment (samarahu) [reply src/rgw/rgw_sal.cc]: Same question as in your other comment

**Review by pritha-srivastava — COMMENTED** [review]
@samarahu : do you intend to remove boost_redis as part of this commit [335e812](https://github.com/pritha-srivastava/ceph/pull/7/commits/335e8122a969c48c78ac88234c8a608438cec42d)? If so, shift this change in another commit and indicate that in your commit message. Also, have you built code to check whether the test file has been actually removed?

**Comment by samarahu** [comment]
> @samarahu : do you intend to remove boost_redis as part of this commit [335e812](https://github.com/pritha-srivastava/ceph/pull/7/commits/335e8122a969c48c78ac88234c8a608438cec42d)? If so, shift this change in another commit and indicate that in your commit message. Also, have you built code to check whether the test file has been actually removed?

@pritha-srivastava I have fixed the `boost_redis` dir removal and confirmed that the test file is removed with the commit introduced in this PR.

File: src/rgw/driver/d4n/d4n_policy.h [hunk 3]
```diff
@@ -77,32 +76,19 @@ class LFUDAPolicy : public CachePolicy {
...
   public:
-    LFUDAPolicy(net::io_context& io_context, rgw::cache::CacheDriver* cacheDriver) : CachePolicy(), io(io_context), cacheDriver{cacheDriver} {
-      conn = std::make_shared<connection>(boost::asio::make_strand(io_context));
-      dir = new BlockDirectory{io};
+    LFUDAPolicy(std::shared_ptr<connection>& conn, rgw::cache::CacheDriver* cacheDriver) : CachePolicy(), 
+                                                                         conn(conn), 
+                                                                         cacheDriver(cacheDriver)
+    {
+      //conn = std::make_shared<connection>(boost::asio::make_strand(io_context));
+      dir = new BlockDirectory{conn};
```
Comment (pritha-srivastava) [inline src/rgw/driver/d4n/d4n_policy.h]: @samarahu : I noticed that you need to update the corresponding test files because of this change.

Comment (samarahu) [reply src/rgw/driver/d4n/d4n_policy.h]: Yes, thank you for reminding me. I meant to do this earlier

**Comment by samarahu** [comment]
Passing run for both small and large object tests: https://pulpito.ceph.com/suriarte-2024-02-22_16:42:00-rgw:d4n-sam-wip-rgw-d4n-distro-default-smithi/

**Pull request closed without merging**